### PR TITLE
[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52865 [quant][graphmode][fx][fp16] Add fp16 support for silu
* #52864 [quant][graphmode][fx][fp16] Add fp16 support for tanh
* #52863 [quant][graphmode][fx][fp16] Add fp16 support for sigmoid
* #52862 [quant][graphmode][fx][fp16] Add fp16 support for layer_norm
* #52811 [quant][graphmode][fx][fp16] Add fp16 support for torch.sum
* #52810 [quant][graphmode][fx][fp16] Add fp16 support for div
* #52809 [quant][graphmode][fx][fp16] Add fp16 support for sub
* #52808 [quant][graphmode][fx] Add support for fp16 bmm pattern
* **#52807 [quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests**
* #52714 [quant][graphmode][fx][fp16] Add fp16 support for {add|mul}{_relu}

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

### CODE REVIEW

File: test/quantization/test_quantize_fx.py [hunk 1]
```diff
@@ -2129,31 +2137,37 @@ def forward(self, x, y):
...
-                OpRelu(is_inplace_op, is_functional_relu, is_scalar),
+                BinaryOpRelu(binary_op, ibinary_op, is_inplace_op, is_functional_relu, is_scalar),
                 data, quant_type, custom_qconfig_dict=custom_qconfig_dict,
-                expected_node_occurrence=node_occurrence,
-                print_debug_info=True)
+                expected_node_occurrence=node_occurrence)
 
 
     @skipIfNoFBGEMM
     def test_add(self):
```
Comment (vkuzo) [inline test/quantization/test_quantize_fx.py:2146]: maybe `_test_binary_op_int8_impl` and `_test_binary_op_float16_impl` could become their own tests, and same for the others?

**Review by vkuzo — APPROVED** [review]

File: test/quantization/test_quantize_fx.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_quantize_fx.py:2228]: hmmm, it's a trade off I think, we can split but then we'll have more tests and it also increases complexity. but i don't have a very strong preference, please let me know if you feel it's worthwhile to make this trade off.

Comment (jerryzh168) [reply test/quantization/test_quantize_fx.py:2228]: will land for now, we can split in a separate PR if needed.
